v2/mongodb-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/mongodb/templates/MongoDbToBigQueryCdc.java

```diff
@@ -68,37 +77,66 @@ public void processElement(ProcessContext context) {
    *
    * @param args Command line arguments to the pipeline.
    */
-  public static void main(String[] args) {
+  public static void main(String[] args)
+      throws ScriptException, IOException, NoSuchMethodException {
 
     Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
     run(options);
   }
 
   /** Pipeline to read data from PubSub and write to MongoDB. */
-  public static boolean run(Options options) {
+  public static boolean run(Options options)
+      throws ScriptException, IOException, NoSuchMethodException {
     options.setStreaming(true);
     Pipeline pipeline = Pipeline.create(options);
-    TableSchema bigquerySchema =
-        MongoDbUtils.getTableFieldSchema(
-            options.getMongoDbUri(),
-            options.getDatabase(),
-            options.getCollection(),
-            options.getUserOption());
-
     String userOption = options.getUserOption();
     String inputOption = options.getInputTopic();
+
+    TableSchema bigquerySchema;
+
+    if (options.getUserOption().equals("UDF")) {
```

Comment:
Same as above, this could check if options.getJavascriptDocumentTransformGcsPath() and options.getJavascriptDocumentTransformFunctionName() are null

Reply:
Removed and refactored